netclient/functions/daemon.go

```diff
@@ -242,6 +242,14 @@ func setupMQTT(cfg *config.ClientConfig, publish bool) mqtt.Client {
 		} else {
 			err = token.Error()
 		}
+		brokerErr := checkBroker(cfg.Server.Server)
+		if brokerErr.Error() == "dns" {
```

Comment:
this code block 246-252 is repeated below, so probably want to convert to a private func and replace it. You also don't check if that err is `nil` like you do below, which could cause a panic.

Reply:
moved logging to checkBroker func